File: firefly-example/src/main/java/com/firefly/example/reactive/coffee/store/service/UserService.java
package com.firefly.example.reactive.coffee.store.service;

import com.firefly.example.reactive.coffee.store.model.User;
import reactor.core.publisher.Mono;

/**
 * @author Pengtao Qiu
 */
public interface UserService {
    Mono<User> getByName(String name);
}

File: firefly-kotlin-ext/src/test/kotlin/com/firefly/kotlin/ext/example/task/management/service/UserService.kt
package com.firefly.kotlin.ext.example.task.management.service

import com.firefly.kotlin.ext.example.task.management.model.User
import com.firefly.kotlin.ext.example.task.management.vo.Request
import com.firefly.kotlin.ext.example.task.management.vo.Response

/**
 * @author Pengtao Qiu
 */
interface UserService {

    suspend fun getUser(request: Request<Long>): Response<User>

    suspend fun listUsers(request: Request<List<Long>>): Response<List<User>>

    suspend fun insert(request: Request<User>): Response<Long>
}